src/mutations/member.ts

```diff
@@ -88,6 +88,46 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
     },
   });
 
+  // suppose you can only edit yourself
+  queryClient.setMutationDefaults(MUTATION_KEYS.UPDATE_PASSWORD, {
+    mutationFn: (payload) =>
+      Api.updatePassword(payload, queryConfig).then((member) => Map(member)),
+    onMutate: async ({ member }) => {
+      // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+      await queryClient.cancelQueries(CURRENT_MEMBER_KEY);
+
+      // Snapshot the previous value
+      const previousMember = queryClient.getQueryData(
+        CURRENT_MEMBER_KEY,
+      ) as Record<Member>;
+
+      // Optimistically update to the new value
+      queryClient.setQueryData(
+        CURRENT_MEMBER_KEY,
+        previousMember.merge(member),
+      );
+
+      // Return a context object with the snapshotted value
+      return { previousMember };
+    },
+    onSuccess: () => {
+      notifier?.({
+        type: editMemberRoutine.SUCCESS,
+        payload: { message: SUCCESS_MESSAGES.EDIT_MEMBER },
+      });
+    },
+    // If the mutation fails, use the context returned from onMutate to roll back
+    onError: (error, _, context) => {
+      notifier?.({ type: editMemberRoutine.FAILURE, payload: { error } });
+      queryClient.setQueryData(CURRENT_MEMBER_KEY, context.previousMember);
+    },
+    // Always refetch after error or success:
+    onSettled: () => {
+      // invalidate all queries
+      queryClient.invalidateQueries(CURRENT_MEMBER_KEY);
```

Comment:
Actually we will never save the password in the frontend (current member won't receive the password anymore) so no need to invalidate the key.